app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java

```diff
@@ -254,13 +255,40 @@ public void onResume()
 
 	private void goToNext()
 	{
-		if (validateFields())
+		if (validateFields() && userDetailsChanged())
 		{
 			YonaAnalytics.createTapEventWithCategory(AnalyticsConstant.SCREEN_EDIT_PROFILE, AnalyticsConstant.SAVE);
 			updateUserProfile();
 		}
+		else
+		{
+			YonaActivity.getActivity().onBackPressed();
+		}
+	}
+
+	private boolean userDetailsChanged()
+	{
+		User loggedInUser = YonaApplication.getEventChangeManager().getDataState().getUser();
+		if (!firstName.getText().toString().equals(loggedInUser.getFirstName()))
```

Comment:
Create a small helper method as follows:
```
private static boolean editTextEquals(EditText editText, String value)
{
	return editText.getText().toString().equals(value);
}
```

Now you can replace all of these if-statements with this:
```
return editTextEquals(firstName, loggedInUser.getFirstName()) ||
	editTextEquals(lastName, loggedInUser.getLastName()) ||
	editTextEquals(nickName, loggedInUser.getNickname()) ||
	editTextEquals(mobileNumber, loggedInUser.getMobileNumber());
```

Reply:
Updated the code.